omegaconf/base.py

```diff
@@ -413,14 +414,36 @@ def _resolve_interpolation_from_parse_tree(
                 raise
             return None
 
-        assert resolved is not None
-        if isinstance(resolved, str):
-            # Result is a string: create a new StringNode for it.
-            return StringNode(
-                value=resolved,
-                key=key,
+        # If the output is not a Node already (e.g., because it is the output of a
+        # custom resolver), then we will need to wrap it within a Node.
+        must_wrap = not isinstance(resolved, Node)
+
+        # If the node is typed, validate (and possibly convert) the result.
+        if isinstance(value, ValueNode) and not isinstance(value, AnyNode):
+            res_value = _get_value(resolved)
+            try:
+                conv_value = value.validate_and_convert(res_value)
+            except ValidationError as e:
+                if throw_on_resolution_failure:
+                    self._format_and_raise(key=key, value=res_value, cause=e)
+                return None
```

Comment:
I think both your comments are related, and we should choose between:
1. Returning `None` on validation error if throw_on_resolution_failure == False, and wrapping the exception into a (subclass of) `InterpolationResolutionError` otherwise
2. Always raising the exception, in which case it should *not* be wrapped inside an `InterpolationResolutionError`. 

This is because it would be weird to say `throw_on_resolution_failure=False` then raise an `InterpolationResolutionError` (although it may depend on what we mean exactly by `throw_on_resolution_failure`: so far I've been interpreting it as catching any exception that happens during interpolation resolution)

Personally I have a slight preference for (1) since it seems more consistent with the rest of the code, but I'll wait to hear your thoughts on this before making any change.

Reply:
I went with (1) in 92757d6 to have something working in a simple and consistent way. I'm open to change it though if you prefer a different approach (but if we decide to change the meaning of `throw_on_resolution_failure`, I'd suggest to do it in a different PR)